src/constants.ts

```diff
@@ -0,0 +1,69 @@
+export declare const extraRpcs: Record<number, string[]>;
+
+export const networkIds: Record<string, number> = {
+  Mainnet: 1,
+  Goerli: 5,
+  Gnosis: 100,
+  Anvil: 31337,
+};
+
+export type Token = {
+  address: string;
+  decimals: number;
+};
+
+export const tokens: Record<number, Record<string, Token>> = {
+  [networkIds.Mainnet]: {
+    DAI: {
+      address: "0x6b175474e89094c44da98b954eedeac495271d0f",
+      decimals: 18,
+    },
+  },
+  [networkIds.Gnosis]: {
+    WXDAI: {
+      address: "0xe91d153e0b41518a2ce8dd3d7944fa863463a97d",
+      decimals: 18,
+    },
+  },
+};
+
+export const LOCAL_HOST = "http://127.0.0.1:8545";
+
+export const networkNames = {
+  [networkIds.Mainnet]: "Ethereum Mainnet",
+  [networkIds.Goerli]: "Goerli Testnet",
+  [networkIds.Gnosis]: "Gnosis Chain",
+  [networkIds.Anvil]: LOCAL_HOST,
+};
+
+export const networkCurrencies: Record<number, { symbol: string; decimals: number }> = {
+  [networkIds.Mainnet]: { symbol: "ETH", decimals: 18 },
+  [networkIds.Goerli]: { symbol: "GoerliETH", decimals: 18 },
+  [networkIds.Gnosis]: { symbol: "XDAI", decimals: 18 },
+  [networkIds.Anvil]: { symbol: "XDAI", decimals: 18 },
+};
+
+export function getNetworkName(networkId?: number) {
+  const networkName = networkNames[networkId as keyof typeof networkNames];
+  if (!networkName) {
+    console.error(`Unknown network ID: ${networkId}`);
+  }
+  return networkName ?? "Unknown Network";
+}
+
+export const networkExplorers: Record<number, string> = {
```

Comment:
Might make more sense to replace `number` by the key of network ids (applies for the other variables as well) to make sure every key is implemented

Reply:
Is what I have now 'too typed'? 

Thought best to pull in all of the network IDs and names but had to restructure their type to match the networkNames structure